apps/lib/opt.c

```diff
@@ -138,6 +138,15 @@ char *opt_progname(const char *argv0)
 }
 #endif
 
+char *opt_appname(const char *arg0)
+{
+    int len = strlen(prog);
+
+    if (arg0 != NULL && strcmp(prog, arg0) != 0)
```

Comment:
However thinking more about it, this won't work properly either if the symlinked app is called with path.

I have an alternative solution - just move the `opt_appname` call from `opt_init` to `main` after the ` /* We assume we've been called as 'openssl cmd' */` comment.

Reply:
It turns out that moving the `opt_appname()` call as you suggested is cleaner but still not sufficient to get rid of the path issue.
For the latter I now use `argv[0] = pname;` in the symlink case such that the command can be found again by `do_cmd()`.